server/controllers/lecture-controller.js

```diff
@@ -168,20 +219,39 @@ exports.cancelBooking = async (req, res) => {
   const lectureId = req.params.lectureid;
   const studentId = req.user && req.user.id;
 
-  // Delete booking lecture in lecture_booking table
   knex("lecture_booking")
     .where("lecture_id", lectureId)
     .andWhere("student_id", studentId)
     .del()
-    .then(() => {
-      res.json({ message: `Booking canceled.` });
+    .then(async (result) => {
+      if (result === 0) throw "No booking to cancel.";
+
+      const [waiting] = await knex("waiting_list")
+        .where("lecture_id", lectureId)
+        .orderBy("booked_at", "asc")
+        .limit(1);
+      if (!waiting) return;
```

Comment:
The problem of the test is maybe this. When the waiting list is empty, it should return res.json({ message: "Booking cancelled." });